src/fun/transform/desugar_with_blocks.rs

```diff
@@ -25,14 +25,14 @@ impl Ctx<'_> {
 }
 
 impl Term {
-  pub fn desugar_do_blocks(
+  pub fn desugar_with_blocks(
     &mut self,
     cur_block: Option<&Name>,
     def_names: &HashSet<Name>,
   ) -> Result<(), String> {
     maybe_grow(|| {
       if let Term::With { typ, bod } = self {
-        bod.desugar_do_blocks(Some(typ), def_names)?;
+        bod.desugar_with_blocks(Some(typ), def_names)?;
         let wrap_ref = Term::r#ref(&format!("{typ}/wrap"));
         // let wrap_ref = if def_names.contains(&wrap_nam) {
```

Comment:
Either we remove this commented code or uncomment it, currently if some type does not have the `wrap` definition but is used inside a `with` block it will fail at the definition_pruning transformation

Reply:
Shouldn't we check for unbound functions at some point? Never noticed we don't do that

Reply:
I would expect this to be catch at the `check_unbound_vars` after `desugar_with_blocks`

Reply:
It's not a variable, so it makes sense. We need a `check_unbound_refs` somewhere near the end of compilation.